File: brute_force/Strings/StringMatchIncrementBy1.java
import java.util.*;

public class StringMatchIncrementBy1 {

    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter Main String :");
        String str1=sc.nextLine();
        System.out.println("Enter Sub String :");
        String str2=sc.nextLine();
        int flag=0,flag1=1,count=0;
        for(int i=0;i<str1.length();i++){
            if(str2.charAt(0)==str1.charAt(i)){
                flag1=0;
                for(int j=0;j<str2.length();j++){
                    if(str2.charAt(j)!=str1.charAt(i+j*2)){
                        flag1=1;
                        break;
                    }
                }
                if(flag1==0){
                    System.out.println("String Matched at Index :"+i);
                    flag=1;
                    count++;
                }
            }
        }

        if(flag==1){
            System.out.println("Match Found :");
        }
        else{
            System.out.println("Match Not Found :");
        }
        System.out.println("String Occured By "+count+" Times");
    }
}

File: brute_force/Strings/StringMatchIncrementBy1.c
#include<stdio.h>
#include<string.h>

int main()
{
    int i,j;
    char str1[100],str2[100];
    int flag=0,c=0,flag1;
    printf("Enter First String :");
    gets(str1);
    printf("Enter Second String :");
    gets(str2);
    int l1=strlen(str1);
    int l2=strlen(str2);
    
    for(i=0;i<l1;i++)
    {
    	if(str2[0]==str1[i])
    	{
    		flag1=0;
    		for(j=0;j<l2;j++)
    		{
    			if(str2[j]!=str1[i+j*2])
    			{
    				flag1=1;
    				break;
    			}
    		}
    		if(flag1==0)
    		{
    			printf("Match Found At Index %d\n",i);
    			c++;
    			flag=1;
    		}
    	}
    }
    
    if(flag!=0)
    {
    	printf("Match Not Found : \n");
    }
    
    printf("Count=%d",c);
      				
}
